public/js/graph/react_graph.js

```diff
@@ -379,6 +389,9 @@ var Graph = React.createClass({
 
     infoBoxMouseClick: function () {
         var infoBox = this.refs.infoBox;
+        this.setState({courseId: infoBox.state.nodeId.substring(0, 6) });
+        this.openModal();
```

Comment:
This results in two separate calls to `setState`, which isn't good. You should modify these two lines to only make one call to `setState`.